src/clients/UBAClient/UBAClientBase.ts

```diff
@@ -64,36 +65,42 @@ export class BaseUBAClient extends BaseAbstractClient {
     return this.retrieveBundleStates(chainId, tokenSymbol).at(-1);
   }
 
+  /**
+   * Returns the most recent bundle state for a chain and token that was created before a given block number.
+   * @param hubPoolBlockNumber The bundle state was proposed at or before this block
+   * @param chainId
+   * @param tokenSymbol
+   * @returns
+   */
+  public retrieveBundleStateForBlock(
+    hubPoolBlockNumber: number,
+    chainId: number,
+    tokenSymbol: string
+  ): UBABundleState | undefined {
+    const sortedDescendingBundleStates = [
+      ...this.retrieveBundleStates(chainId, tokenSymbol).sort(
```

Comment:
QQ: this won't deep copy the bundle states. Is there a reason we're using the spread operator to re-create a new array of bundle states?